articles/cli.md

```diff
@@ -0,0 +1,169 @@
+- No two externalized properties, in the same level, are allowed to share the same name.
+- Additionally, you can externalize multiple properties from a single node.
+- To change the name of the externalized property (usually a good idea), simply edit the "DisplayName" field of the externalized property node.
+
+(![Externalize Property Example 2](~/images/Article_CLI_extern_prop_example_2.png)
+
+##### Passing values from the command line
+To pass values from the command line we use the syntax :
+
+```cmd
+> Bonsai myWorkflow.bonsai -p:<propertyName>=<string>
+```
+You can set properties in nested workflows by using the ```.``` (dot) notation: (e.g. ```<NestedNodeName.propertyName> = <string> ```)
+
+It is worth noting that the same workflow file can be called/run in parallel with different parameters, allowing for easy batch-processing pipelines!
+
+##### Example:
+```cmd
+> :: set a single property
+> bonsai myWorkflow.bonsai --start -p:"myProperty1"="100"
+
+> :: set two distinct properties
+> bonsai myWorkflow.bonsai --start -p:"myProperty1"="100" -p:"myProperty2"="200"
+
+> :: with "--no-editor" mode enabled
+> bonsai myWorkflow.bonsai --no-editor -p:"myProperty1"="100" -p:"myProperty2"="200"
+
+> :: set a property of a nested workflow node
+> bonsai myWorkflow.bonsai --no-editor -p:"myProperty1"="100" -p:"myProperty2"="200" -p:"myNestedNode.myProperty1"="Horizontal"
+```
+___
+&nbsp;
+### Specifying layout files
+As of Bonsai version 2.6.1, it is possible to specify the layout file to run the workflow with, when using the ```--no-editor mode```. This is especially useful when the same workflow is meant to be run several times in parallel (e.g. multiple identical behavior rigs).
+
+The general syntax is:
+
+##### Example:
+```cmd
+
+> :: run the workflow myWorkflow.bonsai with myLayout.bonsai.layout
+> Bonsai myWorkflow.bonsai --no-editor --visualizer-layout:myLayout.bonsai.layout
+```
+
+Currently, the easiest way to generate multiple layouts, for a given script, is to:
+- Launch a workflow with the editor;
+- Place the windows in the desired place;
+- Save the workflow.
+
+This will create a ```*.bonsai.layout``` file with the same name as the workflow which should then be renamed (as to prevent being overwritten) and used later.
+
+It should be noted that ```--visualizer-layout``` can be used in combination with previous flags:
+
+##### Example:
+```cmd
+
+> :: run the workflow with the previous layout while setting property values
+> Bonsai myWorkflow.bonsai --no-editor --visualizer-layout:myLayout.bonsai.layout -p:"myProperty1"="100" -p:"myProperty2"="200" -p:"myNestedNode.myProperty1"="Horizontal"
+
+```
+
+&nbsp;
+
+### Other available flags
+Additional flags are available, and can be used similarly to the ones introduced before. E.g.:
+
+```cmd
+> Bonsai myWorkflow.bonsai --FLAG
+> Bonsai --FLAG
+> Bonsai --FLAG:<"parameterValue">
+```
+
+Currently available flags and a short description:
```

Comment:
Edit for brevity (it's pretty obvious that each flag is listed with a short description): 

"Currently available flags:"